cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/CognitiveServiceBase.scala

```diff
@@ -132,14 +132,78 @@ trait HasSubscriptionKey extends HasServiceParams {
 
   def getSubscriptionKey: String = getScalarParam(subscriptionKey)
 
-  def setSubscriptionKey(v: String): this.type = setScalarParam(subscriptionKey, v)
+  def setSubscriptionKey(v: String): this.type = {
+    println("WARNING: Please use setSubscriptionKey together with setLocation for authentication")
```

Comment:
So we will always emit "WARNING" if they set this? No other way to give this information?

Reply:
Do you have better ideas? Because most users are quite confused what methods should be used together, like many of them want to setUrl but that's not what we want, we should find an elegant way to warn them.

Reply:
you know the code better than me. :)  As a user, I get annoyed by WARNINGS in my logs even if I'm using it right.

My only thoughts:
1. add this info to doc strings? Or are the HasX traits too generic to do that? Can you override doc strings or params if you are specifying a required combination of params needed?
2. validate at transform() time and throw if a proper combination is not set? Is that possible? As a user, this would be useful to me, just to see an error telling me what I did wrong if I didn't set it correctly.

Reply:
Yea I totally agree, it might be annoying. Then I found out that for different cog services the usages are also not consistent, like most of them expects setLocation, but there're some expecting setServiceName instead, and for some other cases like DEP they might have to setUrl explicitly. In this case I'll remove the change in this PR and maybe in the future come up with better documentation in general.

Reply:
If you want to warn, use the official log4j warning APIs please